File: src/main/java/pl/oczadly/kotlin_app/hackyeah2025/demo/controller/DemoController.java
package pl.oczadly.kotlin_app.hackyeah2025.demo.controller;

import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
public class DemoController {

    private final ApiCallDemo apiCallDemo;

    public DemoController(ApiCallDemo apiCallDemo) {
        this.apiCallDemo = apiCallDemo;
    }

    @GetMapping("/demos")
    public List<String> getDemos() {
        return List.of("demo-values");
    }

    @GetMapping("/demos3")
    public String getDemos2() {
        return "new-demo-3";
    }

    @GetMapping("/demo-api-call")
    public String apiCallDemo() {
        return apiCallDemo.makeDemoApiCall();
    }

    @PutMapping("/dynamo-db/{email}")
    public String addEmail(@PathVariable("email") String email) {
        return email;
    }
}

File: src/main/kotlin/pl/oczadly/kotlin_app/hackyeah2025/demo/controller/DemoController.kt
package pl.oczadly.kotlin_app.hackyeah2025.demo.controller

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PutMapping
import org.springframework.web.bind.annotation.RestController


@RestController
class DemoController(private val apiCallDemo: ApiCallDemo) {

    @GetMapping("/demos")
    fun getDemos(): List<String> = listOf("demo-values")

    @GetMapping("/demos3")
    fun getDemos2(): String = "new-demo-3"

    @GetMapping("/demo-api-call")
    fun apiCallDemo(): String? = apiCallDemo.makeDemoApiCall()

    @PutMapping("/dynamo-db/{email}")
    fun addEmail(@PathVariable(name = "email") email: String) = email
}
